<!DOCTYPE html
  PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
  "DTD/xhtml1-strict.dtd">
<html>
  <head>
    <title>API docs for &ldquo;twisted.python.failure.Failure.trap&rdquo;</title>
    <link href="apidocs.css" type="text/css" rel="stylesheet" />
  </head>
  <body>
    <h1 class="function">Function t.p.f.F.trap(self, *errorTypes):</h1>
    <p>
      <span id="part">Part of <a href="twisted.python.failure.Failure.html">twisted.python.failure.Failure</a></span>
      <a></a>
      <a></a>
    </p>
    <div></div>
    <div><p>Trap this failure if its type is in a predetermined list.</p>
<p>This allows you to trap a Failure in an error callback. It will be 
automatically re-raised if it is not a type that you expect.</p>
<p>The reason for having this particular API is because it's very useful 
in Deferred errback chains:</p>
<p>| def _ebFoo(self, failure): | r = failure.trap(Spam, Eggs) | print 
'The Failure is due to either Spam or Eggs!' | if r == Spam: | print 
'Spam did it!' | elif r == Eggs: | print 'Eggs did it!'</p>
If the failure is not a Spam or an Eggs, then the Failure will be 
'passed on' to the next errback.
<table class="fieldTable"><tr class="fieldStart"><td class="fieldName">Parameters</td><td class="fieldArg">errorTypes</td><td>(type: Exception
)</td></tr></table></div>
    
    
        
  </body>
</html>